Error compiling with CLEAR_MEMORY=OFF and OpenSSL

The master branch does not build in windows with visual studio.
Errors: 
```
openssl.c(1934,13): error C2186: '+': illegal operand of type 'void'
openssl.c(2950,13): error C2186: '+': illegal operand of type 'void'
```

- OS: Windows 10
- libssh2 version: master branch
- crypto backend: OpenSSL 3.0.5
- visual studio: 2019/2022

To reproduce, from a visual studio terminal (openssl libs in C:/openssl-x64):

```
mkdir build
cd build
cmake .. ^
 -A x64 ^
 -DCMAKE_INSTALL_PREFIX="C:/libssh2-x64" ^
 -DCMAKE_BUILD_TYPE=Release ^
 -DCRYPTO_BACKEND=OpenSSL ^
 -DBUILD_SHARED_LIBS=OFF ^
 -DOPENSSL_ROOT_DIR=C:/openssl-x64 ^
 -DENABLE_ZLIB_COMPRESSION=OFF ^
 -DBUILD_TESTING=OFF ^
 -DBUILD_EXAMPLES=OFF ^
 -DENABLE_CRYPT_NONE=ON ^
 -DCLEAR_MEMORY=OFF
cmake --build . --config Release --target install
```

--- a/src/misc.h
+++ b/src/misc.h
@@ -40,8 +40,8 @@
 
 #ifdef LIBSSH2_NO_CLEAR_MEMORY
 #define _libssh2_explicit_zero(buf, size) do { \
-                                              (void)buf; \
-                                              (void)size; \
+                                              (void)(buf); \
+                                              (void)(size); \
                                           } while(0)
 #else
 #ifdef WIN32

